components/eam/tools/mkatmsrffile/mkatmsrffile.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python3
+#---------------------------------------------------------------------------------------------------
+'''
+This is a replacement for the legacy gen_domain tool created for CESM. 
+Most legacy functionality is reproduced, with the notable exception of 
+the pole point latitude adjustment needed for the CESM FV grid.
+
+Created April, 2024 by Walter Hannah (LLNL) 
+'''
+#---------------------------------------------------------------------------------------------------
+'''
+The map file used to generate the atmsrf files can be created a few different ways.
+For a typical E3SM configuration we recommend using a conservative, monotone map.
+Here is an example command that can be used to generate one as of NCO version 5.2.2
+
+  SRC_GRID=${DIN_LOC_ROOT}/../mapping/grids/1x1d.nc
+  DST_GRID=${GRID_ROOT}/ne${NE}pg2_scrip.nc
+  MAP_FILE=${MAP_ROOT}/map_1x1_to_ne${NE}pg2_traave.nc
+  ncremap -a traave --src_grd=${SRC_GRID} --dst_grd=${DST_GRID} --map_file=${MAP_FILE}
+
+'''
+#---------------------------------------------------------------------------------------------------
+import datetime, os, numpy as np, xarray as xr, numba, itertools, subprocess as sp
+user, host = os.getenv('USER'), os.getenv('HOST')
+source_code_meta = 'generate_domain_E3SM.py'
+#---------------------------------------------------------------------------------------------------
+class clr:END,RED,GREEN,MAGENTA,CYAN = '\033[0m','\033[31m','\033[32m','\033[35m','\033[36m'
+#---------------------------------------------------------------------------------------------------
+usage = '''
+python mkatmsrffile.py  --map_file <map_file_path> 
+                        --dst_grid <atm_grid_name>
+                        [--output-root <path>]
+                        [--date-stamp <date string>]
```

Comment:
This usage gives an incomplete list of command-line options to run mkatmsrffile.py.

Reply:
fixed.